api/apiclient.go

```diff
@@ -498,11 +505,14 @@ func dialWebsocket(addr, path string, opts DialOpts, tlsConfig *tls.Config, try
 	return try.Start(newWebsocketDialer(cfg, opts))
 }
 
-// newWebsocketDialer returns a function that
-// can be passed to utils/parallel.Try.Start.
-var newWebsocketDialer = createWebsocketDialer
+// newWebsocketDialer is defined as a variable so that it can be overridden
+// for tests.
+var newWebsocketDialer = newWebsocketDialer0
```

Comment:
> I think this is also a decision we've made: moving forward, we fix things as we touch them.

For me "touching" something doesn't just involve a mechanical change to that thing, but actually
involves changing something substantial about the code, otherwise you could say, for example,
that changing an import path "touched" every file that was changed and therefore a PR that
changes an import path needs to fix every file that uses that import.

I have been doing my best to fix any tech debt I've been encountering in code that I change - I
believe strongly in leaving code cleaner than I find it, but I also think that it's really important
to maintain focus in every PR as far as possible (and, yes, I do break that rule
sometimes, usually when rewinding the change to make it independent is going to take lots
of time).